test/Microsoft.ML.Tests/TrainerEstimators/PriorRandomTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.Runtime.RunTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class SimpleEstimatorTests : TestDataPipeBase
+    {
+        private IDataView GetBreastCancerDataviewWithTextColumns()
+        {
+            var dataPath = GetDataPath("breast-cancer.txt");
+            var inputFile = new SimpleFileHandle(Env, dataPath, false, false);
+            return ImportTextData.TextLoader(Env, new ImportTextData.LoaderInput()
+            {
+                Arguments =
+                {
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoader.Column("Label", type: null, 0),
+                        new TextLoader.Column("F1", DataKind.Text, 1),
+                        new TextLoader.Column("F2", DataKind.I4, 2),
+                        new TextLoader.Column("Rest", type: null, new [] { new TextLoader.Range(3, 9) })
+                    }
+                },
+
+                InputFile = inputFile
+            }).Data;
+        }
+
+        public SimpleEstimatorTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void TestEstimatorRandom()
+        {
+            var dataView = GetBreastCancerDataviewWithTextColumns();
+            var pipe = new RandomTrainer(Env);
+
+            // Test only that the schema propagation works.
+            // REVIEW: the save/load is not preserving the full state of the random predictor. This is unfortunate, but we don't care too much at this point.
```

Comment:
Is it still the case?
If yes, can you write down what "full" state are we missing?

Reply:
This comment was written by Pete. I don't know exactly what "full state" refers to. I am going to search more in depth to understand what is causing the problem and file a separate issue if it is needed. The behavior that we observed is that the data produced by the model that was saved + loaded back is different from the data produced by the original model.

---
In reply to: [218520614](https://github.com/dotnet/machinelearning/pull/876#discussion_r218520614) [](ancestors = 218520614)